src/garage/tf/embeddings/base.py

```diff
@@ -0,0 +1,275 @@
+"""Base class for embedding networks in TensorFlow."""
+import abc
+
+import tensorflow as tf
+
+from garage.misc.tensor_utils import flatten_tensors, unflatten_tensors
+
+
+class EmbeddingSpec:
+    """Specification of an embedding network.
+
+    Args:
+        input_space (akro.Space): The input space of the embedding.
+        latent_space (akro.Space): The latent space of the embedding.
+
+    """
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, input_space, latent_space):
+        self.input_space = input_space
+        self.latent_space = latent_space
+
+
+class Embedding(abc.ABC):
```

Comment:
can this be a garage-wide interface, shared with `garage.torch`?

Reply:
I was thinking about the same thing -- put all the common functions inside `garage.BaseModel`? Ultimately all primitives will be a BaseModel / inherits BaseModel.

Reply:
But that's probably hard to be done in this PR